backend/layers/processing/process_logic.py

```diff
@@ -49,41 +49,47 @@ def download_from_s3(self, bucket_name: str, object_key: str, local_filename: st
         self.s3_provider.download_file(bucket_name, object_key, local_filename)
 
     @staticmethod
-    def make_s3_uri(artifact_bucket, bucket_prefix, file_name):
-        return join("s3://", artifact_bucket, bucket_prefix, file_name)
+    def make_s3_uri(artifact_bucket, key_prefix, file_name):
+        return join("s3://", artifact_bucket, key_prefix, file_name)
 
-    def upload(
+    def upload_artifact(
         self,
         file_name: str,
-        bucket_prefix: str,
+        key_prefix: str,
         artifact_bucket: str,
     ) -> str:
         # TODO: make sure that the files are correct
         file_base = basename(file_name)
         self.s3_provider.upload_file(
             file_name,
             artifact_bucket,
-            join(bucket_prefix, file_base),
+            join(key_prefix, file_base),
             extra_args={"ACL": "bucket-owner-full-control"},
         )
-        return ProcessingLogic.make_s3_uri(artifact_bucket, bucket_prefix, file_base)
+        return ProcessingLogic.make_s3_uri(artifact_bucket, key_prefix, file_base)
 
     @logit
     def create_artifact(
         self,
         file_name: str,
         artifact_type: str,
-        bucket_prefix: str,
+        key_prefix: str,
         dataset_id: DatasetVersionId,
         artifact_bucket: str,
         processing_status_key: DatasetStatusKey,
+        datasets_bucket: Optional[str] = None,  # If provided, dataset will be uploaded to this bucket for public access
     ):
         self.update_processing_status(dataset_id, processing_status_key, DatasetConversionStatus.UPLOADING)
-        self.logger.info(f"Uploading [{dataset_id}/{file_name}] to S3 bucket: [{artifact_bucket}].")
         try:
-            s3_uri = self.upload(file_name, bucket_prefix, artifact_bucket)
-            self.logger.info(f"Updating database with  {artifact_type}.")
+            s3_uri = self.upload_artifact(file_name, key_prefix, artifact_bucket)
+            self.logger.info(f"Uploaded [{dataset_id}/{file_name}] to {s3_uri}")
             self.business_logic.add_dataset_artifact(dataset_id, artifact_type, s3_uri)
+            self.logger.info(f"Updated database with {artifact_type}.")
+            if datasets_bucket:
+                datasets_s3_uri = self.s3_provider.upload_file(
```

Comment:
`self.s3_provider.upload_file` does not return the s3_uri, you'll have to build it yourself for the log statement

Reply:
or update the method in s3_provider

Reply:
Also, I think whats leading to the test errors is that, despite the name, extra_args is NOT an optional argument for upload_file (neither in the mock nor the actual s3_provider). Let's update the mock + s3_provider to give it a default arg

Reply:
Great catches. Hadn't had time to investigate fully last night. Resubmitting...